src/annotate/celltypist/test.py

```diff
@@ -0,0 +1,116 @@
+        sc.pp.log1p(data_for_scanpy,
+                    base=None,
+                    layer=None, # use X
+                    copy=False) # allow overwrites in the copy that was made
+        adata.X = data_for_scanpy.X
+        adata.uns['log1p'] = data_for_scanpy.uns['log1p'].copy()
+        input_mudata.mod[modality] = adata
+        transformed_input_mudata_file = random_h5mu_path()
+        input_mudata.write_h5mu(transformed_input_mudata_file)
+        return transformed_input_mudata_file
+    return wrapper
+
+def test_simple_execution(run_component, random_h5mu_path, normalize_log_transform):
+    output_file = random_h5mu_path()
+    input_file_transformed = normalize_log_transform(input_file, "rna")
+
+    run_component([
+        "--input", input_file_transformed,
+        "--reference", reference_file,
+        "--reference_obs_targets", "cell_ontology_class",
+        "--var_reference_gene_names", "ensemblid",
+        "--output", output_file
+    ])
+    
+    assert os.path.exists(output_file), "Output file does not exist"
+    
+    input_mudata = mu.read_h5mu(input_file_transformed)
+    output_mudata = mu.read_h5mu(output_file)
+    
+    assert_annotation_objects_equal(input_mudata.mod["prot"],
+                                    output_mudata.mod["prot"])
+    
+    assert list(output_mudata.mod["rna"].obs.keys()) == ['celltypist_pred',
+                                                         'celltypist_probability']
+    
+    obs_values = output_mudata.mod["rna"].obs["celltypist_probability"]
+    assert all(0 <= value <= 1 for value in obs_values), ".obs at celltypist_probability has values outside the range [0, 1]"
+
+def test_with_model(run_component, random_h5mu_path):
+    output_file = random_h5mu_path()
+
+    run_component([
+        "--input", celltypist_input_file,
+        "--model", model_file,
+        "--reference_obs_targets", "cell_type",
+        "--output", output_file
+    ])
+    
+    assert os.path.exists(output_file), "Output file does not exist"
+    
+    output_mudata = mu.read_h5mu(output_file)
+    
+    assert {'celltypist_pred', 'celltypist_probability'}.issubset(output_mudata.mod["rna"].obs.keys()), "Required keys not found in .obs"
+    
+    obs_values = output_mudata.mod["rna"].obs["celltypist_probability"]
+    assert all(0 <= value <= 1 for value in obs_values), ".obs at celltypist_probability has values outside the range [0, 1]"
+
+def test_fail_check_reference_expression(run_component, random_h5mu_path):
+    output_file = random_h5mu_path()
+
+    with pytest.raises(subprocess.CalledProcessError) as err:
+        run_component([
+            "--input", input_file,
+            "--reference", reference_file,
+            "--var_reference_gene_names", "ensemblid",
+            "--output", output_file,
+            "--check_expression", "True"
+        ])
+    assert re.search(r"ValueError: 🛑 Invalid expression matrix, expect log1p normalized expression to 10000 counts per cell",
```

Comment:
Unfortunatly, we need everything to be in ascii, could you adjust your regex to remove the special character?